src/system_data.jl

```diff
@@ -71,22 +70,36 @@ function SystemData(
     return SystemData(
         components,
         forecast_metadata,
-        validation_descriptors,
         time_series_storage,
-        nothing,
+        validation_descriptors,
         internal,
     )
 end
 
 """
     SystemData(filename::AbstractString)
 
-Deserialize SystemData from a JSON file.
+Construct SystemData from a JSON file.
 """
 function SystemData(filename::AbstractString)
     return from_json(SystemData, filename)
 end
 
+"""Deserializes a SystemData from a JSON file."""
+function from_json(::Type{SystemData}, filename::String)
+    # File paths in the JSON are relative. Temporarily change to this directory in order
+    # to find all dependent files.
+    orig_dir = pwd()
+    cd(dirname(filename))
+    try
+        return open(filename) do io
+            from_json(io, SystemData)
+        end
+    finally
+        cd(orig_dir)
+    end
+end
+
```

Comment:
Are de-serializing from a folder or a JSON file? If from a folder shouldn't we check that the input is a folder?